CMakeLists.txt

```diff
@@ -1,35 +1,97 @@
-file(GLOB files_decomp decomp_2d.f90
-	               log.f90
-                       glassman.f90)
-include_directories(${CMAKE_SOURCE_DIR}/decomp2d)
-
-#if(${FFT_Choice} MATCHES "generic")
-#  file(GLOB files_fft fft_generic.f90)
-#endif(${FFT_Choice} MATCHES "generic")
-
-if(FFTW_FOUND)
-  message (STATUS "Compiling using FFTW3")
-  file(GLOB files_fft fft_fftw3.f90)
-else(FFTW_FOUND)
-  message (STATUS "Compiling using Generic FFT")
-  file(GLOB files_fft fft_generic.f90)
-endif(FFTW_FOUND)
-
-
-set(SRCFILES ${files_decomp} ${files_fft})
-
-add_library(decomp2d STATIC ${SRCFILES})
-if (MPI_FOUND)
-  target_link_libraries(decomp2d PRIVATE MPI::MPI_Fortran)
-endif (MPI_FOUND)
-
-if(FFTW_FOUND)
-  target_include_directories(decomp2d PRIVATE ${FFTW_INCLUDE_DIRS})
-  #target_include_directories(decomp2d PRIVATE ${PKG_FFTW_LIBRARY_DIRS})
-endif(FFTW_FOUND)
-
-install(TARGETS decomp2d
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} 
-)
+cmake_minimum_required(VERSION 3.12)                        
+cmake_policy(SET CMP0074 NEW)
+
+project(2decomp LANGUAGES Fortran)
+enable_testing()
+
+
+set(BUILD_TARGET "mpi" CACHE STRING "Target for acceleration (mpi (default) or gpu)")
+set_property(CACHE BUILD_TARGET PROPERTY STRINGS mpi gpu)
+
+if (BUILD_TARGET MATCHES "gpu")
+  option(ENABLE_OPENACC "Allow user to activate/deactivate OpenACC support" ON)
+  option(ENABLE_CUDA "Allow user to activate/deactivate CUDA support" ON)
+  option(ENABLE_MANAGED "Allow user to activate/deactivate automatic memory managment from NVHPC" OFF)
+  option(ENABLE_NCCL "Allow user to activate/deactivate Collective Comunication NCCL" OFF)
+
+  if (ENABLE_CUDA)
+    message(STATUS "Before enable CUDA")
+    enable_language(CUDA)
+    message(STATUS "After enable CUDA")
+  endif()
+
+endif(BUILD_TARGET MATCHES "gpu")
+
+set (ENABLE_PROFILER "OFF" CACHE STRING "Activate/select the profiler")
+set_property(CACHE ENABLE_PROFILER PROPERTY STRINGS OFF caliper)
+option(ENABLE_INPLACE "Activate the profiler" OFF)
```

Comment:
Doc string should be "Activate in place FFT" ?

Reply:
Yes I think will make more sense

Reply:
Resolved by db1f9db